integration/setup/setup.go

```diff
@@ -30,7 +30,7 @@ const (
 	credentialNamespace   = "giantswarm"
 )
 
-func HostPeerVPC(c *awsclient.Client, g *framework.Guest, h *framework.Host) error {
+func HostPeerVPC(c *awsclient.Client, g *framework.Guest, h *framework.Host) (string, error) {
```

Comment:
When I look at that I feel like we should create `Setup` struct which would take all the clients and frameworks and provide install methods.

Reply:
@MarcelMue's idea from https://github.com/giantswarm/giantswarm/issues/3878 is to have a single config in `e2e-harness/pkg/framework` that takes in all clients.

Config will be optional so we only validate if we use host or guest functionality. This should let us align the setup across all projects. I think it's a good approach and could help here.